java/src/main/native/src/RmmJni.cpp

```diff
@@ -75,48 +75,32 @@ class tracking_resource_adaptor final : public base_tracking_resource_adaptor {
   Upstream *get_wrapped_resource() { return resource; }
 
   std::size_t get_total_allocated() override {
-    std::lock_guard<std::mutex> lock(size_map_mutex);
-    return total_allocated;
+    return total_allocated.load();
   }
 
 private:
   Upstream *const resource;
   std::size_t const size_align;
-  std::size_t total_allocated{0};
-
-  // map and associated lock to track memory sizes by address
-  // TODO: This should be removed when rmm::alloc and rmm::free are removed and the size parameter
-  //       for do_deallocate can be trusted. If map and mutex are removed then total_allocated
-  //       should be updated to be atomic.
-  std::unordered_map<void *, std::size_t> size_map{};
```

Comment:
The include of the `unordered_map` header should be able to be removed after this change since the type is no longer referenced in this file.

Reply:
done